docs/faq.rst

```diff
@@ -210,8 +210,19 @@ be used used to bind a socket locally::
     async with aiohttp.ClientSession(connector=conn) as session:
         ...
 
-.. seealso:: :class:`aiohttp.TCPConnector` and ``local_addr`` parameter.
+If your OS supports it, you may also bind directly to the interface ::
+
+    try:
+        conn = aiohttp.TCPConnector(network_interface="eth0")
+        async with aiohttp.ClientSession(connector=conn) as session:
+            ...
+    except RuntimeError:
+        # fallback to local_addr as the runtime kernel is unsupported.
+        conn = aiohttp.TCPConnector(local_addr=('127.0.0.1', 0))
+        ...
```

Comment:
```suggestion
If your OS supports it, you may also bind directly to the interface ::

    conn = aiohttp.TCPConnector(network_interface="eth0")
    async with aiohttp.ClientSession(connector=conn) as session:
        ...

To prefer a network interface, it is possible to catch the `RuntimeError` on creation and create a fallback:

    try:
        conn = aiohttp.TCPConnector(network_interface="eth0")
    except RuntimeError:
        # fallback to local_addr as the runtime kernel is unsupported.
        conn = aiohttp.TCPConnector(local_addr=('127.0.0.1', 0))

    async with aiohttp.ClientSession(connector=conn) as session:
        ...
```

Reply:
@Dreamsorcerer you need to specify a role for the `RuntimeError` reference in Markdown. It may look like this:
```rst
:exc:`RuntimeError`
```

Without it, the role is `any` which essentially matches the first found thing and if it breaks, it may be non-trivial to an inexperienced person to debug.

Reply:
@Mostafa-Mahdi explanation on Sam's suggestion: try-except blocks should always wrap only one instruction for which you'd like to process an exception. Wrapping more may cause unintended side effects that might appear in the future silently and would be hard to notice and track down.

Reply:
The original also duplicates the 2nd line unnecessarily.